static/js/upload.js

```diff
@@ -13,187 +17,223 @@ exports.feature_check = function (upload_button) {
+    exports.get_item("send_status_message", config).text(message);
+};
+
+exports.upload_files = function (uppy, config, files) {
+    if (files.length === 0) {
+        uppy.cancelAll();
+        exports.hide_upload_status(config);
+        return;
+    }
+    if (page_params.max_file_upload_size === 0) {
+        exports.show_error_message(config, i18n.t('File and image uploads have been disabled for this organization.'));
+        return;
+    }
+    exports.get_item("send_button", config).attr("disabled", "");
+    exports.get_item("send_status", config).addClass("alert-info").removeClass("alert-error").show();
+    exports.get_item("send_status_message", config).html($("<p>").text(i18n.t("Uploading…")));
+    exports.get_item("send_status_close_button", config).one('click', function () {
+        uppy.cancelAll();
+        setTimeout(function () {
+            exports.hide_upload_status(config);
+        }, 500);
+    });
+
+    for (const file of files) {
+        try {
+            compose_ui.insert_syntax_and_focus("[Uploading " + file.name + "…]()", exports.get_item("textarea", config));
+            compose_ui.autosize_textarea();
+            uppy.addFile({
+                source: exports.get_item("source", config),
+                name: file.name,
+                type: file.type,
+                data: file,
+            });
+        } catch (error) {
+            // Errors are handled by info-visible and upload-error event callbacks.
         }
+    }
+};
 
-        switch (error_code) {
-        case 'BrowserNotSupported':
-            msg = i18n.t("File upload is not yet available for your browser.");
-            break;
-        case 'TooManyFiles':
-            msg = i18n.t("Unable to upload that many files at once.");
-            break;
-        case 'FileTooLarge':
-            if (page_params.max_file_upload_size > 0) {
-                // sanitization not needed as the file name is not potentially parsed as HTML, etc.
-                const context = {
-                    file_name: file.name,
-                    file_size: page_params.max_file_upload_size,
-                };
-                msg = i18n.t('"__file_name__" was too large; the maximum file size is __file_size__MB.',
-                             context);
-            } else {
-                // If uploading files has been disabled.
-                msg = i18n.t('File and image uploads have been disabled for this organization.');
-            }
-            break;
-        case 413: // HTTP status "Request Entity Too Large"
-            msg = i18n.t("Sorry, the file was too large.");
-            break;
-        case 400: {
-            const server_message = server_response && server_response.msg;
-            msg = server_message || i18n.t("An unknown error occurred.");
-            break;
+exports.setup_upload = function (config) {
+    const uppy = Uppy({
+        debug: true,
+        autoProceed: true,
```

Comment:
Can you talk about your thinking around these settings choices?

Reply:
I forgot to set the `debug` paramater to `false`. Fixed it.

If `autoProceed ` is set to `False` we have to call a function called `upload` after `addFile` function is called. I didn't find any reason to set it to False. If autoProceed === true, Uppy will begin uploading automatically when files are added.

https://uppy.io/docs/uppy/#autoProceed-false
https://uppy.io/docs/uppy/#uppy-addFile-fileObject